File: src/main/java/com/github/squirrelgrip/meter/exception/InvalidVersionException.java
package com.github.squirrelgrip.meter.exception;

import com.github.squirrelgrip.meter.domain.ContextKey;

import java.util.Map;

public class InvalidVersionException extends MeterReaderException {
    public InvalidVersionException(String recordIndicator, Map<ContextKey, Object> metaData) {
        super("Record " + recordIndicator + " cannot be used in " + metaData.get(ContextKey.VERSION), metaData);
    }
}

File: src/main/kotlin/com/github/squirrelgrip/meter/exception/InvalidVersionException.kt
package com.github.squirrelgrip.meter.exception

import com.github.squirrelgrip.meter.domain.MeterContext
import com.github.squirrelgrip.meter.domain.Version

class InvalidVersionException(recordIndicator: String, requiredVersion: Version, context: MeterContext) :
    MeterReaderException(
        "Record $recordIndicator requires version $requiredVersion but ${context.version} is specified",
        context
    )
